src/settings/VersionHistory/VersionHistory.js

```diff
@@ -0,0 +1,106 @@
+import { useIntl } from 'react-intl';
+import { Field } from 'react-final-form';
+
+import {
+  TitleManager,
+  useCallout,
+} from '@folio/stripes/core';
+import {
+  Row,
+  Col,
+  Select,
+  Label,
+  Layout,
+  LoadingPane,
+} from '@folio/stripes/components';
+import { ConfigFinalForm } from '@folio/stripes/smart-components';
+
+import { useAuditSettings } from '../../queries';
+import { VERSION_HISTORY_PAGE_SIZE_SETTING } from '../../constants';
+
+const VersionHistory = () => {
+  const intl = useIntl();
+  const callout = useCallout();
+
+  const {
+    settings,
+    updateSetting,
+    isLoading,
+    isError,
+  } = useAuditSettings();
+
+  const fieldNames = {
+    PAGE_SIZE: 'pageSize',
+  };
+
+  const pageSizeSettings = settings?.find(setting => setting.key === VERSION_HISTORY_PAGE_SIZE_SETTING);
+
+  const pageSizeOptions = [10, 25, 50, 75, 100].map(value => ({ label: value, value }));
+
+  const handleSubmit = async formValues => {
+    try {
+      await updateSetting({
+        body: {
+          ...pageSizeSettings,
+          value: formValues[fieldNames.PAGE_SIZE],
+        },
+        settingKey: VERSION_HISTORY_PAGE_SIZE_SETTING,
+      });
+
+      callout.sendCallout({
+        message: intl.formatMessage({ id: 'stripes-smart-components.cm.success' }),
+      });
+    } catch (e) {
+      callout.sendCallout({
+        type: 'error',
+        message: intl.formatMessage({ id: 'ui-marc-authorities.error.defaultSaveError' }),
+      });
+    }
+  };
+
+  if (isLoading) {
+    return <LoadingPane />;
+  }
+
+  if (isError) {
+    callout.sendCallout({
+      type: 'error',
+      message: intl.formatMessage({ id: 'ui-marc-authorities.settings.versionHistory.loadError' }),
+    });
+
+    return null;
+  }
+
+  return (
+    <Layout className="full">
+      <TitleManager record={intl.formatMessage({ id: 'ui-marc-authorities.settings.versionHistory.pane.title' })}>
+        <ConfigFinalForm
+          label={intl.formatMessage({ id: 'ui-marc-authorities.settings.versionHistory.pane.title' })}
+          initialValues={{
+            [fieldNames.PAGE_SIZE]: pageSizeSettings?.value,
+          }}
+          onSubmit={handleSubmit}
+        >
+          <Row>
+            <Col xs={12}>
+              <Label htmlFor={fieldNames.PAGE_SIZE}>
+                {intl.formatMessage({ id: 'ui-marc-authorities.settings.versionHistory.field.pageSize' })}
+              </Label>
+            </Col>
+            <Col xs={4}>
+              <Field
+                id={fieldNames.PAGE_SIZE}
+                name={fieldNames.PAGE_SIZE}
+                dataOptions={pageSizeOptions}
+                component={Select}
+                parse={v => parseInt(v, 10)}
+              />
+            </Col>
```

Comment:
Could you describe your logic here, please? I see two rows on the mockup - one with a label, and another with input. You render it within one row, but two columns one of which takes up the entire row width. Why wouldn't you just render Select with a label?

Reply:
@mariia-aloshyna the label will wrap to the next line due to its length, but now I extended the Select component width.